core/Cargo.toml

```diff
@@ -18,15 +18,10 @@ categories.workspace = true
 workspace = true
 
 [features]
-default = ["cli", "telemetry"]
+default = ["telemetry"]
 
 # Support lightweight telemetry, including diagnostics
 telemetry = []
-# Support the included CLI
-cli = []
-# Support developer-specific telemetry.
-# Should not be enabled on production builds.
-dev-telemetry = ["telemetry", "iroha_telemetry/dev-telemetry"]
 # Support Prometheus metrics. See https://prometheus.io/.
 expensive-telemetry = ["iroha_telemetry/metric-instrumentation"]
```

Comment:
I think this isn't used and can be removed

Reply:
There are many `cfg`s in `iroha_telemetry_derive`'s generated code relying on this feature. I wonder why it isn't used by `iroha` crate, and how Prometheus works without it...

Reply:
There is still a mess around telemetry features...

- In Torii, `telemetry` enables `/version` (why it is telemetry?), `/status` (again, why?), and `/metrics` (Prometheus data) endpoints;
- In Iroha itself, `telemetry` enables an WebSocket service which submits a telemetry events to some endpoint. `dev_telemetry` extends previous case to writing to a file, and also enables tokio console in `iroha_logger`.
- `Metrics` in Sumeragi in `iroha_core`, which are served as Prometheus metrics, are collected anyway, independent from enabled features
- `iroha_core` has `telemetry` feature, which doesn't enable anything else, and isn't used inside by the crate
- purpose of `iroha_telemetry/metric-instrumentation` isn't really clear

Reply:
@mversic, should we address it now, or separately, or never?

Reply:
let 's open a ticket

Reply:
Opened: #4386